lib/the_little_thinkers_space_web/controllers/profile_controller.ex

```diff
@@ -3,41 +3,55 @@ defmodule TheLittleThinkersSpaceWeb.ProfileController do
 
   alias TheLittleThinkersSpace.Accounts
   alias TheLittleThinkersSpace.Accounts.Profile
+  alias TheLittleThinkersSpace.Accounts.User
 
-  def index(conn, _params) do
-    profiles = Accounts.list_profiles()
-    render(conn, "index.html", profiles: profiles)
+  plug :authenticate_user when action in [:index, :new, :create, :show, :edit, :update, :delete]
+  action_fallback TheLittleThinkersSpaceWeb.FallbackController
+
+  def index(conn, _params, current_user) do
+    with :ok <- Bodyguard.permit(Profile, :index, current_user) do
+      profiles = Accounts.list_profiles()
+      render(conn, "index.html", profiles: profiles)
+    end
   end
 
-  def new(conn, _params) do
+  def new(conn, _params, _current_user) do
```

Comment:
I would be tempted to do this, event though it is probably unnecessary.

```suggestion
  def new(conn, _params, %User{}) do
```

Reply:
Don't feel like you need to make this change.

Reply:
I'm not feeling it ;)